app/src/main/java/com/spascoding/taskycourse/feature_agenda_screen/presentation/agenda/AgendaViewModel.kt

```diff
@@ -0,0 +1,35 @@
+package com.spascoding.taskycourse.feature_agenda_screen.presentation.agenda
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.spascoding.taskycourse.feature_register_screen.data.local.model.UserInfoManager
+import com.spascoding.taskycourse.feature_register_screen.data.local.model.userInfo
+import com.spascoding.taskycourse.feature_register_screen.domain.use_case.AuthenticationUseCases
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+@HiltViewModel
+class AgendaViewModel @Inject constructor(
+    private val authenticationUseCases: AuthenticationUseCases,
+    private val userInfoManager: UserInfoManager,
+) : ViewModel() {
+
+    fun onEvent(event: AgendaEvent) {
+        when (event) {
+            is AgendaEvent.LogoutAction -> {
+                viewModelScope.launch {
+                    userInfoManager.userInfoFlow.collect { userInfo ->
+                        authenticationUseCases.logoutUser.invoke(userInfo.accessToken) {
+                            event.onSuccess.invoke()
+                            viewModelScope.launch {
+                                userInfoManager.clearUserInfo()
+                            }
```

Comment:
Here it could happen that you pop the back stack after logging out, which will clear the ViewModel and therefore cancel all coroutines in `viewModelScope` leading to this line not being executed if clearing the VM is faster than clearing user info. To fix this you could either invoke the success lambda after clearing the prefs or you execute this in an application wide coroutine scope.